lib/edit-engine/addon/routes/application.js

```diff
@@ -0,0 +1,28 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class ApplicationRoute extends Route {
+  @service content;
+
+  model(params) {
+    let model;
+    if (params.dataset_id) {
+      model = this.content.fetchDataset(params.dataset_id);
+    }
+    if (params.document_id) {
+      model = this.content.fetchDocument(params.document_id)
+    }
+    return model;
+  }
+
+  renderTemplate(applicationController, model) {
+    const template = model.type;
+    const controller = this.controllerFor(model.type);
+    this.render(template, {
+      into: 'application',
+      controller,
+      model
+    });
+  }
```

Comment:
This is the key: As far as I can tell there is no way to mount a particular route of an engine at a particular route in the host. So we will always land at the edit-engine's application route.

If we then transition to a child route we get urls like `documents/abc123/edit/document`. Yuck.

But what we can do is leverage `render`/`renderTemplate` to decide what controller to render based on the model we got.